### PR TITLE
Frontend changes for 036

### PR DESCRIPTION
**Pull request by MattDHill**
- Remove warning related to using Tor over http
- Accommodate new interfaces paradigm for LAN port forwarding upgrade

### CODE REVIEW

File: web/projects/ui/src/app/pages/apps-routes/app-interfaces/app-interfaces.page.ts [hunk 1]
```diff
@@ -126,3 +100,65 @@ export class AppInterfacesItemComponent {
...
   }
 }
+
+function getAddresses(
+  serviceInterface: ServiceInterfaceWithHostInfo,
+): MappedAddress[] {
+  const host = serviceInterface.hostInfo
+  const addressInfo = serviceInterface.addressInfo
+  const username = addressInfo.username ? addressInfo.username + '@' : ''
+  const suffix = addressInfo.suffix ? '/' + addressInfo.suffix : ''
```
Comment (dr-bonez) [inline web/projects/ui/src/app/pages/apps-routes/app-interfaces/app-interfaces.page.ts:110]: I think the suffix might already include a /

File: web/projects/ui/src/app/pages/apps-routes/app-interfaces/app-interfaces.page.ts [hunk 2]
```diff
@@ -126,3 +100,65 @@ export class AppInterfacesItemComponent {
...
+      } else {
+        name = h.hostname.kind
+        hostname =
+          h.hostname.kind === 'domain'
+            ? `${h.hostname.subdomain}.${h.hostname.domain}`
+            : h.hostname.value
+      }
+
+      if (h.hostname.sslPort) {
+        const port = h.hostname.sslPort === 443 ? '' : `:${h.hostname.sslPort}`
```
Comment (dr-bonez) [inline web/projects/ui/src/app/pages/apps-routes/app-interfaces/app-interfaces.page.ts:138]: Don't hardcode 443. Only erase the port if it matches the defaultPort for a KnownProtocol. So look up the scheme in knownProtocols, and check the port against KnownProtocol.defaultPort

File: web/projects/ui/src/app/services/config.service.ts [hunk 3]
```diff
@@ -57,24 +56,60 @@ export class ConfigService {
...
+
+    if (host.kind === 'multi') {
+      const onionHostname = host.hostnames.find(
+        h => h.kind === 'onion',
+      ) as HostnameInfoOnion
+
+      if (this.isTor() && onionHostname) {
+        url.hostname = onionHostname.hostname.value
+      } else {
+        const ipHostname = host.hostnames.find(
```
Comment (dr-bonez) [inline web/projects/ui/src/app/services/config.service.ts:88]: Just taking the first ip hostname you find? Not trying to match it to the address they're currently on?

**Review by dr-bonez — CHANGES_REQUESTED** [review]

File: web/projects/ui/src/app/pages/apps-routes/app-interfaces/app-interfaces.page.ts (hunk 2 above)
Comment (MattDHill) [reply web/projects/ui/src/app/pages/apps-routes/app-interfaces/app-interfaces.page.ts:138]: These are always UIs

File: web/projects/ui/src/app/pages/apps-routes/app-interfaces/app-interfaces.page.ts (hunk 1 above)
Comment (MattDHill) [reply web/projects/ui/src/app/pages/apps-routes/app-interfaces/app-interfaces.page.ts:110]: Why is that. Are you expecting package devs to write it, or is the BE adding it universally to the string?

File: web/projects/ui/src/app/services/config.service.ts (hunk 3 above)
Comment (MattDHill) [reply web/projects/ui/src/app/services/config.service.ts:88]: Look at line 94. Just trying to figure out if they have an IP hostname. This won't work once we have domains.

File: web/projects/ui/src/app/pages/apps-routes/app-interfaces/app-interfaces.page.ts (hunk 1 above)
Comment (dr-bonez) [reply web/projects/ui/src/app/pages/apps-routes/app-interfaces/app-interfaces.page.ts:110]: It gets built by the sdk from other components, and that function adds a leading / if appropriate

File: web/projects/ui/src/app/services/config.service.ts (hunk 3 above)
Comment (dr-bonez) [reply web/projects/ui/src/app/services/config.service.ts:88]: Hmm, ok, I guess that works

File: web/projects/ui/src/app/pages/apps-routes/app-interfaces/app-interfaces.page.ts (hunk 2 above)
Comment (dr-bonez) [reply web/projects/ui/src/app/pages/apps-routes/app-interfaces/app-interfaces.page.ts:138]: Then I suppose it technically would work, though I'd document the reliance on all uis being http. Personally I'd abstract url construction to a different function and handle it there though cause you'll need it for displaying interface urls too

**Review by dr-bonez — APPROVED** [review]

**Pull request merged**